app/addons/documents/routes-documents.js

```diff
@@ -181,11 +194,14 @@ function (app, FauxtonAPI, BaseRoute, Documents, Changes, ChangesActions, DocEdi
       },
 
       cleanup: function () {
+
+        // to ensure garbage collection on the React components
+        this.removeComponent('#dashboard-lower-content');
```

Comment:
This shouldn't be needed. the `FauxtonAPI.RouteObject.prototype.cleanup.apply(this);` should do that for you